src/main/java/de/fraunhofer/iosb/ilt/sta/model/builder/api/AbstractMultiDatastreamBuilder.java

```diff
@@ -0,0 +1,148 @@
+package de.fraunhofer.iosb.ilt.sta.model.builder.api;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import de.fraunhofer.iosb.ilt.sta.model.MultiDatastream;
+import de.fraunhofer.iosb.ilt.sta.model.Observation;
+import de.fraunhofer.iosb.ilt.sta.model.ObservedProperty;
+import de.fraunhofer.iosb.ilt.sta.model.Sensor;
+import de.fraunhofer.iosb.ilt.sta.model.Thing;
+import de.fraunhofer.iosb.ilt.sta.model.ext.UnitOfMeasurement;
+import org.geojson.Polygon;
+import org.threeten.extra.Interval;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Base class for any {@link AbstractEntityBuilder} of {@link MultiDatastream}
+ *
+ * @param <U> the type of the concrete class that extends this {@link AbstractMultiDatastreamBuilder}
+ * @author Aurelien Bourdon
+ */
+public abstract class AbstractMultiDatastreamBuilder<U extends AbstractMultiDatastreamBuilder<U>> extends AbstractEntityBuilder<MultiDatastream, U> {
+
+    @Override
+    protected MultiDatastream newBuildingInstance() {
+        return new MultiDatastream();
+    }
+
+    @Override
+    public MultiDatastream build() {
+        // A MultiDatastream is always set by the ValueCode#OM_ComplexObservation observation type
+        getBuildingInstance().setObservationType(ValueCode.OM_ComplexObservation.getValue());
+        return super.build();
+    }
+
+    @Override
+    protected MultiDatastream getBuildingInstance() {
+        return new MultiDatastream();
```

Comment:
Is that supposed to be here? It'll return a new MultiDatastream each time...

Reply:
Oups, well seen ;-) I fix it.